tests/test_assembly_arg_str.py

```diff
@@ -0,0 +1,13 @@
+from xdsl.dialects.arm.assembly import assembly_arg_str
+from xdsl.dialects.arm.attrs import LabelAttr
+from xdsl.dialects.arm.register import IntRegisterType
+
+
+def test_assembly_arg_str_ARMRegister():
+    arg = IntRegisterType("x0")
+    assert assembly_arg_str(arg) == "x0"
+
+
+def test_assembly_arg_str_LabelAttr():
+    arg = LabelAttr("main")
+    assert assembly_arg_str(arg) == "main"
```

Comment:
not really sure how I would test for other SSAValue types and invalid ones

Reply:
Maybe use `TestSSAValue` with a `TestType` type? if you search for these in the repo, you'll find examples.
You're trying to cover the `else` when the SSAValue is not of `ARMRegisterType` right?

Reply:
yes, that's exactly what I was looking for, thank you @compor. I have updated the test now